planner/type_join.go

```diff
@@ -261,7 +261,7 @@ func (p *Planner) makeTypeJoinOne(
 
 	subTypeField, subTypeFieldNameFound := subTypeCol.Description().GetFieldByRelation(
 		subTypeFieldDesc.RelationName,
-		parent.collection.Name(),
+		parent.collection.Name().Value(),
```

Comment:
> question: Why have you skipped `HasValue()` here before calling `Value()`. If IRRC the value returned is invalid unless `HasValue() == true`

question: Same as above

Reply:
If we are joining a collection to another, both will have names.